### PR TITLE
Neruwhine can no longer give the hypnosis brain trauma

### LINKED ISSUE #63188: You can have a trauma of "nothing"
Reporting client version: 514.1568

## Round ID:
[173975](https://scrubby.melonmesa.com/round/173975)

## Testmerges:
- [Mouse Entered Profiling](https://www.github.com/tgstation/tgstation/pull/63098)
- [Fixes Massive Radio Overtime, Implements a Spatial Grid System for Faster Searching Over Areas](https://www.github.com/tgstation/tgstation/pull/61422)
- [pAI TGUI conversion](https://www.github.com/tgstation/tgstation/pull/62916)

## Reproduction:
image
Saw it in the round before this one on Sybil as well, only thing I could guess at to explain this was that the person with the trauma was hypnotized, but had hypnotized victim disabled in prefs. There were no associated runtimes.

### PR DESCRIPTION
**Pull request by FernandoJ8**
## About The Pull Request

Fixes #63188

## Why It's Good For The Game

Bugfix, plus we already removed obsessed from that list cause it gives people antag and hypnosis does the same. 

## Changelog

:cl:
fix: Removes hypnosis from the list of traumas neruwhine can cause, fixing a pair of related bugs.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm [hunk 1]
```diff
@@ -631,6 +631,7 @@ Basically, we fill the time between now and 2s from now with hands based off the
 	var/traumalist = subtypesof(/datum/brain_trauma)
 	traumalist -= /datum/brain_trauma/severe/split_personality //Uses a ghost, I don't want to use a ghost for a temp thing.
 	traumalist -= /datum/brain_trauma/special/obsessed //Sets the owner as an antag - I presume this will lead to problems, so we'll remove it
+	traumalist -= /datum/brain_trauma/hypnosis //Same reason as obsessed, plus a bug makes it remain even after the neurowhine purges and then turn into "nothing" on the med reading upon a second application
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm:634]: at this point just make a list this looks silly

Comment (FernandoJ8) [reply code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm]: Will do

**Comment by ATH1909** [comment]
on a tangential note, why are split personalities blocked from being rolled by neruwhine? we could just handle them the same way that neruwhine handles imaginary friends instead, no?

**Comment by SuperNovaa41** [comment]
> on a tangential note, why are split personalities blocked from being rolled by neruwhine? we could just handle them the same way that neruwhine handles imaginary friends instead, no?

> // Split personality uses a ghost, I don't want to use a ghost for a temp thing

It is a temporary effect. Getting a ghost role for an extremely short amount of time is pretty stupid, and a waste of a ghost poll.

**Comment by moocowswag** [comment]
> It is a temporary effect. Getting a ghost role for an extremely short amount of time is pretty stupid, and a waste of a ghost poll.

Athath was refering to the fact that neruwhine`s code suggests that imaginary friends are permanent (dont go away after neruwhine is used up) and he asked why not make split personality like that to.

(This is actually false though, on multiple instances i`ve played with neruwhine I`ve had temporary/brief imaginary friends which does exactly what you are trying to avoid, a waste of a ghost poll, so there is something that probably needs to be addressed)

File: code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm [hunk 2]
```diff
@@ -629,8 +629,8 @@ Basically, we fill the time between now and 2s from now with hands based off the
 	if(!(DT_PROB(creation_purity*10, delta_time)))
 		return
 	var/traumalist = subtypesof(/datum/brain_trauma)
-	traumalist -= /datum/brain_trauma/severe/split_personality //Uses a ghost, I don't want to use a ghost for a temp thing.
-	traumalist -= /datum/brain_trauma/special/obsessed //Sets the owner as an antag - I presume this will lead to problems, so we'll remove it
+	var/list/forbiddentraumas = list(/datum/brain_trauma/severe/split_personality, /datum/brain_trauma/special/obsessed, /datum/brain_trauma/hypnosis) // Split personality uses a ghost, I don't want to use a ghost for a temp thing; Obsessed sets the owner as an antag - I presume this will lead to problems, so we'll remove it; Hypnosis, same reason as obsessed, plus a bug makes it remain even after the neurowhine purges and then turn into "nothing" on the med reading upon a second application
+	traumalist -= forbiddentraumas
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm:633]: I think I prefer removal and reasoning on the same line, split up like before. Was the target cleanliness?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm (hunk 2 above)
Comment (FernandoJ8) [reply code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm:633]: It was, but you're right. Ordering it like this is a bit unwieldy, I'll split it up

**Pull request merged**